src/ai/backend/manager/registry.py

```diff
@@ -3149,19 +3151,21 @@ async def _update() -> None:
                 )
 
             # Update the mapping of kernel images to agents.
-            loaded_images = msgpack.unpackb(zlib.decompress(agent_info["images"]))
+            images = msgpack.unpackb(zlib.decompress(agent_info["images"]))
+            image_canonicals = set(img_info[0] for img_info in images)
+            prev_image_canonicals = self.agent_installed_images.get(agent_id, set())
 
             async def _pipe_builder(r: Redis):
                 pipe = r.pipeline()
-                for image, _ in loaded_images:
-                    try:
-                        await pipe.sadd(ImageRef.parse_image_str(image, "*").canonical, agent_id)
-                    except ValueError:
-                        # Skip opaque (non-Backend.AI) image.
-                        continue
```

Comment:
I intentionally omitted the logic to skip image caching when a ValueError occurs while parsing the image string since I think excluding the loaded images sent from the agent from only redis caching is meaningless.